code/game/objects/items/reagent_containers/autoinjectors.dm

```diff
@@ -10,6 +10,13 @@
 	volume = 30
 	list_reagents = list(/datum/reagent/consumable/sodiumchloride = 30)
 
+/obj/item/reagent_containers/hypospray/autoinjector/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Autoinjector", name)
+		ui.open()
+		return TRUE // Or else the other UI on reagent_containers/interact will open
```

Comment:
Just remove the ui opening from it's interact, it's redundant anyway. Add a line in the examine that you can right click the hypo to change the injection rate as well

Reply:
This is for reagent bottles.